include/zephyr/drivers/crc.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2024 Brill Power Ltd.
+ * Copyright (c) 2025 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief CRC public API header file.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CRC_H
+#define ZEPHYR_INCLUDE_DRIVERS_CRC_H
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/crc.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief CRC driver APIs
+ * @defgroup crc_interface CRC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+#define CRC_FLAG_REVERSE_INPUT     BIT(0)
```

Comment:
API should be fully documented, i.e all macros, struct and enum fields, etc. Thanks!